deltacat/utils/daft.py

```diff
@@ -78,10 +78,36 @@ def daft_s3_file_to_table(
     logger.debug(f"Time to read S3 object from {s3_url} into daft table: {latency}s")
 
     if kwargs.get("schema") is not None:
-        schema = kwargs["schema"]
+        input_schema = kwargs["schema"]
         if include_columns is not None:
-            schema = pa.schema([schema.field(col) for col in include_columns])
-        daft_schema = Schema.from_pyarrow_schema(schema)
-        return table.cast_to_schema(daft_schema).to_arrow()
+            input_schema = pa.schema(
```

Comment:
Should we copy schema metadata as well?

Reply:
Currently our behavior is to use the metadata available on the `table_schema` (i.e. in the Parquet file)

This is the logic inherited from the existing pyarrow impl: https://github.com/ray-project/deltacat/blob/main/deltacat/utils/pyarrow.py#L331

Should we instead use the metadata available on the schema that is provided by DeltaCat?

Reply:
This shouldn't really matter because we don't use schema metadata today. However, input schema metadata contains descriptions of the columns while table schema metadata doesn't. Hence, the former is generally right but will require more memory.

Reply:
Ok I will switch our logic to use the input schema metadata instead